Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1062,15 +1067,43 @@ public extension Purchases {
     }
 #endif
 
-#if os(iOS) || VISION_OS
+    #if (os(iOS) || VISION_OS) && !targetEnvironment(macCatalyst)
 
     @available(iOS 14.0, *)
     @available(watchOS, unavailable)
     @available(tvOS, unavailable)
     @available(macOS, unavailable)
+    @available(macCatalyst 16.0, *)
+    @objc func presentCodeRedemptionSheet(
+        uiWindowScene: UIWindowScene? = nil
+    ) async throws {
+        try await self.presentOfferCodeRedemptionSheet(uiWindowScene: uiWindowScene)
+    }
+
+    @available(iOS 14.0, *)
     @available(macCatalyst, unavailable)
-    @objc func presentCodeRedemptionSheet() {
-        self.paymentQueueWrapper.paymentQueueWrapperType.presentCodeRedemptionSheet()
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(macOS, unavailable)
+    internal func presentOfferCodeRedemptionSheet(uiWindowScene: UIWindowScene?) async throws {
+        var windowScene = uiWindowScene
+        if windowScene == nil {
+            windowScene = try await systemInfo.currentWindowScene
+        }
+
+        guard let windowScene else {
+            Logger.error(Strings.storeKit.error_displaying_offer_code_redemption_sheet_no_window_scene)
```

Comment:
It might be a little awkward that this is possible, but I thought it was important to keep the provided window scene optional for two reasons:
1. So that developers using SwiftUI can use easily the API
2. To keep the new `Purchases.presentCodeRedemptionSheet` API as similar as possible to the deprecated one